File: apps/api/aqapi/auth/domain/value_object/user_id.py
from uuid import UUID
from aqapi.core.domain.value_object.base_value_object import BaseValueObject


class UserId(BaseValueObject):
    """ユーザーのIDを表す値オブジェクト"""

    def __init__(self, value: UUID):
        self._value = value
        super().__init__()

    def _validate(self) -> None:
        if not isinstance(self._value, UUID):
            raise ValueError("IDはUUIDでなければなりません。")
            

File: packages/backend/src/features/auth/value-object/userId.ts
import { LocaleString } from "../../../core/messages/localeString";
import { BaseValueObject } from "../../../core/value-object/baseValueObject";
import { z } from 'zod';

/**
 * UserIdのZodスキーマ
 */
export const UserIdSchema = z.string();

/**
 * ユーザーIDを表す値オブジェクト(UUID)
 */
export class UserId extends BaseValueObject<string, typeof UserIdSchema> {
  constructor(value: string) {
    super({ value });
  }

  protected validate(): void {
    // UUID形式の基本的なバリデーション
    if (!this.value || this.value.trim() === '') {
      throw new Error('User ID is required');
    }
  }

  protected get valueName(): LocaleString {
    return new LocaleString({
      ja: "ユーザーID",
      en: "User ID"
    });
  }

  /**
   * Zodスキーマから新しいUserIdインスタンスを作成
   * @param data Zodスキーマに準拠したデータ
   */
  static fromZodData(data: z.infer<typeof UserIdSchema>): UserId {
    return new UserId(data);
  }
}
